### PR TITLE
Update compileSdkVersion for Android examples

### PR DESCRIPTION
**Pull request by stuartmorgan**
Flutter master now requires that Android applications use
compileSdkVersion 31 in order to build.

This required updating the Kotlin and gradle versions as well.

And that in turn requires Java 11, which means the outdated
Dockerfile needs to be updated. This updates to use the same
base image (and formatter) that flutter/plugins now uses.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/rfw/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,12 @@
 ## NEXT
 
+* Updates example compileSdkVersion to 31.
```
Comment (camsim99) [inline packages/rfw/CHANGELOG.md:3]: Could you clarify which example app corresponds to which update?

**Review by camsim99 — APPROVED** [review]

File: packages/rfw/CHANGELOG.md (hunk 1 above)
Comment (stuartmorgan) [reply packages/rfw/CHANGELOG.md:3]: Sorry, I had fixed this double-NEXT locally when I uploaded, but apparently on the wrong branch as part of messing up my local branches :P

**Comment by stuartmorgan** [comment]
This has ballooned quite a bit:
- the compile SDK updated required gradle updates
- that in turn required a Java update
- which requires a Dockerfile update
- which affect web tests, and also clang-format

These dependent changes had already been done incrementally in flutter/plugins, but not here, and since the failure is out-of-band we have to fix them all at once now :(

**Comment by stuartmorgan** [comment]
Closing as obsolete; I've moved the parts of this I still want to do over to https://github.com/flutter/packages/pull/537

**Pull request closed without merging**